src/components/settings/connection-settings-card.tsx

```diff
@@ -130,15 +130,15 @@ const DecryptionSpinner = styled(Icon).attrs(() => ({
 `;
 
 const isValidHost = (host: string | undefined): boolean =>
```

Comment:
These two methods are also used for the no-proxy hostnames and the client certificate hostname, where using a username & password doesn't make any sense, so this isn't quite right

I think it's better if instead of changing all hostname validation, we make two separate functions (`isValidProxyHost` and `validateProxyHost`) and then we change the logic just there, without changing these existing methods. Does that make sense?

Reply:
yea it makes sense, I see this was the reason why you made separate function